### PR TITLE
Draw the scene into a depth texture as well

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

File: escher/gl/depth_buffer.cc [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+  glTexImage2D(GL_TEXTURE_2D, 0, GL_DEPTH_COMPONENT, size.width(),
+               size.height(), 0, GL_DEPTH_COMPONENT, GL_UNSIGNED_INT, nullptr);
+
+  glBindFramebuffer(GL_FRAMEBUFFER, frame_buffer_.id());
+  glFramebufferTexture2D(GL_FRAMEBUFFER, GL_DEPTH_ATTACHMENT, GL_TEXTURE_2D,
+                         texture_.id(), 0);
+
+  GLenum status = glCheckFramebufferStatus(GL_FRAMEBUFFER);
```
Comment (j9brown) [inline escher/gl/depth_buffer.cc:38]: This may block.  Only do it in debug mode?

File: escher/gl/depth_buffer.cc [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  if (!frame_buffer_)
+    frame_buffer_.Generate();
+  texture_.Generate();
+
+  glBindTexture(GL_TEXTURE_2D, texture_.id());
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+  glTexImage2D(GL_TEXTURE_2D, 0, GL_DEPTH_COMPONENT, size.width(),
```
Comment (j9brown) [inline escher/gl/depth_buffer.cc:30]: Do we need to give it content here?  I assume we'll glClear it later.

File: escher/gl/unique_frame_buffer.cc [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+UniqueFrameBuffer::~UniqueFrameBuffer() {
+  Clear();
+}
+
+void UniqueFrameBuffer::Generate() {
+  Clear();
+  glGenFramebuffers(1, &id_);
+}
+
+void UniqueFrameBuffer::Clear() {
```
Comment (j9brown) [inline escher/gl/unique_frame_buffer.cc]: Clear is ambiguous here.  Consider reset or release.

File: escher/gl/depth_buffer.cc (hunk 2 above)
Comment (abarth) [inline escher/gl/depth_buffer.cc:30]: We just need to allocate the memory.  We'll glClear it.

File: escher/gl/unique_frame_buffer.cc (hunk 3 above)
Comment (abarth) [inline escher/gl/unique_frame_buffer.cc]: Done.

File: escher/gl/depth_buffer.cc (hunk 1 above)
Comment (abarth) [inline escher/gl/depth_buffer.cc:38]: Done.

**Comment by abarth** [comment]
PTAL

**Pull request merged**